tests/functional/conftest.py

```diff
@@ -35,13 +36,17 @@ def app(pyramid_app, db_engine):
     return TestApp(pyramid_app)
 
 
-@pytest.fixture
+@pytest.fixture(autouse=True)
 def db_session(db_engine):
     """Get a standalone database session for preparing database state."""
 
     conn = db_engine.connect()
     session = SESSION(bind=conn)
 
-    yield session
+    factories.set_sqlalchemy_session(session, persistence="commit")
 
-    session.close()
+    try:
+        yield session
+    finally:
+        factories.clear_sqlalchemy_session()
+        session.close()
```

Comment:
Had to add Factory Boy support to the functional tests, the same as we've done for the unit tests